core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -94,8 +88,20 @@ class LogSegment private[log] (val log: FileRecords,
   private var rollingBasedTimestamp: Option[Long] = None
 
   /* The maximum timestamp we see so far */
-  @volatile private var maxTimestampSoFar: Long = timeIndex.lastEntry.timestamp
-  @volatile private var offsetOfMaxTimestamp: Long = timeIndex.lastEntry.offset
+  @volatile private var _maxTimestampSoFar: Long = -1L
```

Comment:
When there is no time index (old message format), the default timestamp is -1. So, we will need a different default value.

Reply:
I see. I will use Option[Long] instead.